src/hooks/cakeVault/useGetVaultUserInfo.ts

```diff
@@ -1,37 +1,21 @@
-import { useEffect, useState } from 'react'
 import BigNumber from 'bignumber.js'
-import { useWeb3React } from '@web3-react/core'
-import { BIG_ZERO } from 'utils/bigNumber'
-import { useCakeVaultContract } from 'hooks/useContract'
+import { useCakeVault } from 'state/hooks'
 
-const useGetVaultUserInfo = (lastUpdated?: number) => {
-  const { account } = useWeb3React()
-  const cakeVaultContract = useCakeVaultContract()
-  const [userInfo, setUserInfo] = useState({
-    shares: BIG_ZERO,
-    cakeAtLastUserAction: BIG_ZERO,
-    lastDepositedTime: '',
-    lastUserActionTime: '',
-  })
+const useGetVaultUserInfo = () => {
+  const {
+    userData: {
+      isLoading,
+      userShares: userSharesAsString,
+      cakeAtLastUserAction: cakeAtLastUserActionAsString,
+      lastDepositedTime,
+      lastUserActionTime,
+    },
+  } = useCakeVault()
 
-  useEffect(() => {
-    //   user-specific vault contract fetches
-    const fetchUserVaultInfo = async () => {
-      const userContractInfo = await cakeVaultContract.methods.userInfo(account).call()
-      setUserInfo({
-        shares: new BigNumber(userContractInfo.shares),
-        cakeAtLastUserAction: new BigNumber(userContractInfo.cakeAtLastUserAction),
-        lastDepositedTime: userContractInfo.lastDepositedTime,
-        lastUserActionTime: userContractInfo.lastUserActionTime,
-      })
-    }
+  const userShares = new BigNumber(userSharesAsString)
+  const cakeAtLastUserAction = new BigNumber(cakeAtLastUserActionAsString)
```

Comment:
This hook now simply parses serialised BigNumbers from redux, and returns them as fully-formed BigNumbers.

It saves us doing this sort of thing each time the state is accessed:

```js
const { userShares: userSharesAsString } = useCakeVault()
  const userShares = new BigNumber(userSharesAsString)
```

Reply:
I think you have the right idea. BigNumbers should be stored as `toJSON` and converted back by their respective accessor hooks.

Reply:
This is now all concentrated within the `useCakeVault` state hook, much neater.